SMLHelper/Patchers/WorldEntityDatabasePatcher.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace SMLHelper.V2.Patchers
+{
+    using Harmony;
+    using System;
+    using UWE;
+
+    internal class WorldEntityDatabasePatcher
+    {
+        internal static SelfCheckingDictionary<string, WorldEntityInfo> CustomWorldEntityInfos = new SelfCheckingDictionary<string, WorldEntityInfo>("CustomWorldEntityInfo");
+
+        internal static void Patch(HarmonyInstance harmony)
+        {
+            harmony.Patch(AccessTools.Method(typeof(WorldEntityDatabase), "TryGetInfo"),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(WorldEntityDatabasePatcher), "Prefix")));
+        }
+
+        private static bool Prefix(string classId, ref WorldEntityInfo info, ref bool __result)
+        {
+            foreach(var entry in CustomWorldEntityInfos)
+            {
+                if(entry.Key == classId)
+                {
+                    __result = true;
+                    info = entry.Value;
+                    return false;
```

Comment:
Why are we prefix patching a dictionary lookup instead of just adding our custom entries to the `infos` dictionary directly?

Reply:
I tried patching the constructor as there is no "initialize" method, so to speak, which didn't work for some reason (maybe the constructor was already called at that point?). And anyways, the dictionary is never looked up by the game's code directly, all calls are made to TryGetInfo.

Reply:
Yeah, doing a postfix patch on the constructor would be the better option since there are no Initialize methods here.